internal/cmd/versions.go

```diff
@@ -25,14 +31,73 @@ import (
 func NewVersionsCmd(o *globals.GlobalOpts) *cli.Command {
 	return &cli.Command{
 		Name:  "versions",
-		Usage: "List available Envoy versions",
+		Usage: "List Envoy versions",
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name:    "all",
+				Aliases: []string{"a"},
+				Usage:   "Show all versions including ones not yet installed",
+			}},
 		Action: func(c *cli.Context) error {
-			m, err := envoy.GetEnvoyVersions(o.EnvoyVersionsURL, o.UserAgent)
-			if err != nil {
+			vd := map[string]string{}
+			if err := addInstalledVersions(vd, o.HomeDir); err != nil {
 				return err
 			}
-			envoy.PrintVersions(m, globals.CurrentPlatform, c.App.Writer)
+			if c.Bool("all") {
+				if ev, err := envoy.GetEnvoyVersions(o.EnvoyVersionsURL, o.UserAgent); err != nil {
+					return err
+				} else if err := envoy.AddVersions(vd, ev.Versions, globals.CurrentPlatform); err != nil {
+					return err
+				}
+			}
+			if len(vd) == 0 {
+				fmt.Fprintln(c.App.Writer, "No Envoy versions, yet") //nolint
+			} else {
+				printVersions(vd, c.App.Writer)
+			}
 			return nil
 		},
 	}
 }
+
+// addInstalledVersions adds installed Envoy versions
+func addInstalledVersions(vd map[string]string, envoyHome string) error {
+	files, err := os.ReadDir(filepath.Join(envoyHome, "versions"))
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+		if i, err := f.Info(); f.IsDir() && err == nil {
+			vd[f.Name()] = i.ModTime().Format("2006-01-02")
```

Comment:
I would prefer using `[]struct{ version string , releaseDate string}` to map[string]string.. with the latter we have to always keep in mind that, like, a key is a Envoy's versions and a value is release date.. WDYT? In anyway, it is hard to drive the meaning of key/values just by looking at the variable name `vd` at the first glance tbh

Reply:
yeah actually you defined `versionReleaseDate` struct in printVersions, so why not here as well?

Reply:
I used a map so that you don't end up with duplicate rows on merge..

Reply:
I will do it internal to merge

Reply:
nice!

Reply:
done now. thanks! it is nicer at the end